src/components/dialog/index.tsx

```diff
@@ -283,7 +288,7 @@ class Dialog extends Component<DialogProps, DialogState> {
         transparent
         visible={modalVisibility}
         animationType={'none'}
-        onBackgroundPress={this.hideDialogView}
+        onBackgroundPress={onBackgroundPress}
         onRequestClose={this.hideDialogView}
```

Comment:
It feels like this should be affected as well (`onRequestClose` is the back button in Android)